### PR TITLE
Do not install jupyter in conda environment.

### PR DESCRIPTION
**Pull request by yakutovicha**
Previously, everything was installed in the base conda environment,
however, this created some problems in coupling jupyter with AiiDA
which requires tornado to be lower than 5.0.0. To fix this problem
this PR installs jupyter in a 'root' environment, but configures it
to use the ipykernel from the base conda environment.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -34,24 +37,33 @@ chown -R root:root /opt/pseudos/;                                  \
...
-RUN jupyter serverextension enable --sys-prefix --py nbserverproxy
+    'fileupload==0.1.5'            \
+    'nbserverproxy==0.8.8'         \
+    'appmode-aiidalab==0.5.0.1'    \
+    'notebook==5.7.8'              \
+    'nglview'                      \
+    'voila'
+
+# Enable extensions.
+RUN /usr/local/bin/jupyter serverextension enable --sys-prefix --py nbserverproxy
```
Comment (ltalirz) [inline Dockerfile:54]: Just to improve readability
```suggestion
RUN /usr/local/bin/jupyter serverextension enable --py --sys-prefix nbserverproxy
```

File: Dockerfile [hunk 2]
```diff
@@ -34,24 +37,33 @@ chown -R root:root /opt/pseudos/;                                  \
...
+RUN /usr/local/bin/jupyter nbextension     enable --py --sys-prefix appmode
+RUN /usr/local/bin/jupyter serverextension enable --py --sys-prefix appmode
+RUN /usr/local/bin/jupyter nbextension enable nglview --py --sys-prefix
+# TODO: delete, when https://github.com/aiidalab/aiidalab-widgets-base/issues/31 is fixed
+# the fileupload extension also needs to be "installed".
+RUN /usr/local/bin/jupyter nbextension install --sys-prefix --py fileupload
```
Comment (ltalirz) [inline Dockerfile:60]:
```suggestion
RUN /usr/local/bin/jupyter nbextension install --py --sys-prefix fileupload
```

File: Dockerfile [hunk 3]
```diff
@@ -34,24 +37,33 @@ chown -R root:root /opt/pseudos/;                                  \
...
     'jupyterlab==0.35.4'           \
-    'nbserverproxy==0.8.8'
-
-# Enable nbserverproxy extension.
-RUN jupyter serverextension enable --sys-prefix --py nbserverproxy
+    'fileupload==0.1.5'            \
+    'nbserverproxy==0.8.8'         \
+    'appmode-aiidalab==0.5.0.1'    \
+    'notebook==5.7.8'              \
+    'nglview'                      \
```
Comment (ltalirz) [inline Dockerfile:50]: just out of curiosity: why is this needed here?

File: Dockerfile [hunk 4]
```diff
@@ -34,24 +37,33 @@ chown -R root:root /opt/pseudos/;                                  \
...
+    'voila'
+
+# Enable extensions.
+RUN /usr/local/bin/jupyter serverextension enable --sys-prefix --py nbserverproxy
+RUN /usr/local/bin/jupyter nbextension     enable --py --sys-prefix appmode
```
Comment (ltalirz) [inline Dockerfile:55]: also here: why is this needed now, but wasn't needed before?

File: Dockerfile [hunk 5]
```diff
@@ -34,24 +37,33 @@ chown -R root:root /opt/pseudos/;                                  \
...
 # Enables better integration with Jupyter Hub.
 # https://jupyterlab.readthedocs.io/en/stable/user/jupyterhub.html#further-integration
 # Takes about 3 minutes and 20 seconds.
 RUN jupyter labextension install @jupyterlab/hub-extension
 
-# TODO: delete, when https://github.com/aiidalab/aiidalab-widgets-base/issues/31 is fixed
-# the fileupload extension also needs to be "installed".
-RUN jupyter nbextension install --sys-prefix --py fileupload
-
 # Install jupyterlab theme.
```
Comment (ltalirz) [inline Dockerfile:67]: by the way, as long as we aren't using it, I think it's perfectly fine to comment out anything related to the jupyterlab
(not sure whether the latest voilà stuff is starting to use it?)

File: opt/prepare-aiidalab.sh [hunk 6]
```diff
@@ -91,3 +91,6 @@ if [ ! -e /home/${SYSTEM_USER}/apps ]; then
   git checkout ${AIIDALAB_DEFAULT_GIT_BRANCH}
  cd -
 fi
+
+# Activate the right python kernel.
+python -m ipykernel install --user
```
Comment (ltalirz) [inline opt/prepare-aiidalab.sh:96]: can this go in the dockerfile with the other python-setup-related commands?

File: Dockerfile (hunk 3 above)
Comment (yakutovicha) [reply Dockerfile:50]: this was a workaround to be able to run https://github.com/aiidalab/aiidalab-docker-stack/blob/821e78c00677e5f6139375da1c633cc643524657/Dockerfile#L57
which, in turn, was necessary to have `nglview` working.

Comment (yakutovicha) [reply Dockerfile:50]: To be clear: I am aware that this is a hack. Not sure, though, what is the better way to achieve the same. One needs to play around with it.

File: Dockerfile (hunk 4 above)
Comment (yakutovicha) [reply Dockerfile:55]: Well, without this appmode won't work. Not sure why, though.

The docker container that Ole had created actually contained that line. His container was based on the same principles we are now basing the new container on - keeping Jupyter stuff in a separate python environment.

When I was moving stuff to AiiDA 1.0, I removed it because I moved everything in the same Python3 environment. Now I am moving things back just to give another evidence that _history repeats itself_ ;)

File: opt/prepare-aiidalab.sh (hunk 6 above)
Comment (yakutovicha) [reply opt/prepare-aiidalab.sh:96]: It was originally in the Docker file, but then I moved it here. The reason was this `--user` option. But it is not clear whether the option is at all needed. I will now try it out without `--user`.

File: Dockerfile (hunk 5 above)
Comment (yakutovicha) [reply Dockerfile:67]: I think Dou might start using it, so let's keep it.

**Comment by yakutovicha** [comment]
> Could you please also let me know the easiest instructions for testing it locally?
> Can I just follow the readme?

No, I think we should add an option to build things locally, in the current implementation `./run_aiidalab.sh` pulls the image from the Docker Hub. 

you should do:
```bash
./build.sh
docker run -d -p 9999:8888 -e JUPYTER_TOKEN="PUTH_YOUR_TOKEN_HERE" -v /Users/yakutovicha/aiidalab_test:/home/aiida aiidalab-docker-stack:develop
```

File: opt/prepare-aiidalab.sh (hunk 6 above)
Comment (yakutovicha) [reply opt/prepare-aiidalab.sh]: worked, so I moved it back to the docker file in ac4d402

File: Dockerfile (hunk 3 above)
Comment (yakutovicha) [reply Dockerfile:52]: done
